main/models.py

```diff
@@ -203,6 +203,48 @@ def _warn_no_funding(self) -> None | str:
             return "No funding defined for the project."
         return None
 
+    def _warn_phase_lifetime(self) -> None | str:
+        """Warns if the phases don't cover the project lifetime."""
+        # get phases for project id
+        phases_query = ProjectPhase.objects.filter(project__name=self.name)
+
+        # required for mypy as Project.start_date can be blank and null
+        assert self.start_date is not None
+
+        # check start overlaps
+        starts = phases_query.filter(start_date=self.start_date).exists()
+
+        # iterate through phase end dates checking alignment with starts or project end.
+        span = False
+        for query in phases_query:
+            expected_start_date = query.end_date + timedelta(days=1)
+            span = (
+                phases_query.filter(start_date=expected_start_date).exists()
+                or self.end_date == expected_start_date
+            )
+
+        # do the check
+        if not (starts or span):
+            return "Phases do not span project lifetime."
+        return None
+
+    def _warn_wrong_days_sum(self) -> None | str:
+        """Warns if the phases do not sum to the total working days for the project."""
+        project_days = self.total_working_days
+
+        # get query for project name
+        phases_query = ProjectPhase.objects.filter(project__name=self.name)
+        phase_days = 0
+        for phase in phases_query:
+            phase_days += phase.days
```

Comment:
This can be more succinct and efficient as:
```suggestion
        phase_days = sum(phase.days for phase in ProjectPhase.objects.filter(project__name=self.name))
```